client.js

```diff
@@ -80,7 +80,7 @@ Discord.Client = class Client extends Discord.Client {
 						dumped = true;
 						const cache = this.dumpCache();
 						const sessions = this.dumpSessions();
-						if(options.hotReload.onExit) {
+						if(options.hotReload?.onExit) {
```

Comment:
this shouldnt be needed since we already know options.hotReload is truthy in this scope

Reply:
But we don't know if it is an object it could be a bool. Users can set hotReload to be true and let us handle all of the logic *or* provide their own object with sessions, cache and onExit function